content/advent-2018/go-devops.md

```diff
@@ -0,0 +1,190 @@
+Testing, profiling (CPU and memory) and benchmarking are all built-in, so there's no need for learning a micro language, new commands or use new tools, and there are some nice tools for results visualisations.
+
+#### One Binary to Rule Them All
+Go is statically linked, meaning there's no need for external libraries, copy dependencies or worry for imports. All the code and its dependencies are in the binary, so that's all you need to distribute. And as a purely homogenous environment it's not dependent on language versions and releases.
+
+#### Cross Compilation
+Having everything in the binary makes things simple, being able to cross compile it makes things simple even in organizations where everyone have their own setup: in order for a binary to be supported on the different operating systems all it takes is setting the 2 environment variables: $GOOS, $GOARCH. No need for a virtual environment, a package manager or managing dependencies. This is a great feature for CLIs, and indeed some of the most popular ones are using it: etcdctl, kubectl and docker.
+
+Here's a partial list:
+
+|  $GOOS  | $GOARCH |       OS       |
+|---------|---------|----------------|
+| darwin  | 386     | 32 bit MacOSX  |
+| darwin  | amd64   | 64 bit MacOSX  |
+| linux   | 386     | 32 bit Linux   |
+| linux   | amd64   | 64 bit Linux   |
+| linux   | arm     | RISC Linux     |
+| windows | 386     | 32 bit Windows |
+| windows | amd64   | 64 bit Windows |
+
+#### Composition, Not Inheritance
+Staying out of inheritance confusion.
+
+#### The Standard Librart
+Many of the packages in the standard library are the building blocks of the Ops toolbox like handling web services with different protocols like HTTP/HTTP2,
+and file processing: path, download, open, process, time, json, regex, etc. No need for keeping track what is the currently standard package or external software for the bsaic operations, or switching if it gets deprecated. As these are all in the standard library - they are all lofficially maintained, well documented and in use across all the developers. 
+While the std lib is not big but is wisely composed of packages that proof useful for an OPS/SRE person
+
+### Reliable
+
+#### Pointers Exist, Pointer Arithmetic Doesn’t
+Stayin safe!
+
+#### Error Handling
+Planning clearly for errors and acting upon them as values, rather than having error exceptions, makes the execution smoother.as values.
+
+#### Open Source
+Go is backed by an amazing community, companies who are using the language for years now, and by industry giants like Google, Microsoft, Apple and more (did you know that SpaceX is using Go?). So by now, the language is here to stay.
+
+#### Data Types
+Go is Type safe and strongly typed, which means that string operations on an int cannot happen, because it will be caight by the compiler.
```

Comment:
```suggestion
Go is Type safe and strongly typed, which means that string operations on an int cannot happen because the compiler will catch it.
```